### PR TITLE
Alternate implementation w/multiple error codes.

### PR DESCRIPTION
**Pull request by serac**

### CODE REVIEW

**Comment by dfish3r** [comment]
Not API compatible.
```
***! MODIFIED CLASS: PUBLIC org.passay.RuleResultDetail  (not serializable)
        ===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
        ---! REMOVED FIELD: PROTECTED(-) FINAL(-) java.lang.String errorCode
```

File: src/main/java/org/passay/RuleResultDetail.java [hunk 1]
```diff
@@ -30,19 +31,52 @@ public RuleResultDetail(final String code, final Map<String, Object> params)
...
+
+
+
+  /**
+   * Creates a new rule result detail.
+   *
+   * @param  params  error details.
+   * @param  codes  One or more error codes. Codes MUST be provided in order of decreasing specificity.
+   */
+  public RuleResultDetail(final Map<String, Object> params, final String... codes)
```
Comment (dfish3r) [inline src/main/java/org/passay/RuleResultDetail.java]: I'd probably forgo varargs here just to keep symmetry with the other map. Plus it just feels like the codes are the first order concern and should come first.

File: src/main/java/org/passay/RuleResultDetail.java [hunk 2]
```diff
@@ -71,6 +105,6 @@ public String getErrorCode()
   @Override
   public String toString()
   {
-    return String.format("%s:%s", errorCode, parameters);
+    return String.format("%s:%s", Arrays.asList(errorCodes), parameters);
```
Comment (dfish3r) [inline src/main/java/org/passay/RuleResultDetail.java]: `Arrays#toString` is probably better suited.

File: src/main/java/org/passay/RuleResultDetail.java (hunk 1 above)
Comment (serac) [reply src/main/java/org/passay/RuleResultDetail.java]: Fixed.

File: src/main/java/org/passay/RuleResultDetail.java (hunk 2 above)
Comment (serac) [reply src/main/java/org/passay/RuleResultDetail.java]: Fixed.

**Comment by serac** [comment]
Addressed code review feedback. I can't comment on your API incompatible feedback. It's obviously API compatible in this sense of it has the same public API, but maybe you mean it _ABI_ incompatible, which I can see. That doesn't seem like an issue, but you know more about this than I do so I defer to your judgement.

**Comment by dfish3r** [comment]
I've seen this called 'API Binary Compatibility'. Whatever it's called, it means you don't have to recompile your code to use the next revision of library. Which is not the case here. This change bumps the version to 1.6.0, can you update the pom in this PR?

**Comment by serac** [comment]
Bumped version

**Pull request merged**